src/Inventory.ts

```diff
@@ -28,128 +49,35 @@ function getClosestChest (bot: Bot, chestLocations: Vec3[]): Vec3 | null {
   return chest
 }
 
-export function emptyInventoryIfFull (bot: Bot, chestLocations: Vec3[], itemFilter: ItemFilter, cb: Callback): void {
-  if (bot.inventory.emptySlotCount() > 0) {
-    cb()
-    return
-  }
-
-  emptyInventory(bot, chestLocations, itemFilter, cb)
-}
-
-export function emptyInventory (bot: Bot, chestLocations: Vec3[], itemFilter: ItemFilter, cb: Callback): void {
-  if (chestLocations.length === 0) {
-    cb(error('NoChests', 'There are no defined chest locations!'))
-    return
-  }
-
-  // Shallow clone so we can safely remove chests from the list that are full.
-  chestLocations = [...chestLocations]
-
-  const tryNextChest = (): void => {
-    const chest = getClosestChest(bot, chestLocations)
-
-    if (chest == null) {
-      cb(error('NoChests', 'All chests are full.'))
-      return
-    }
-
-    tryEmptyInventory(bot, chest, itemFilter, (err: Error | undefined, hasRemaining: boolean): void => {
-      if (err != null) {
-        cb(err)
-        return
-      }
-
-      if (!hasRemaining) {
-        cb()
-        return
-      }
-
-      tryNextChest()
-    })
-  }
+async function tryEmptyInventory (options: InventoryOptions, location: Vec3): Promise<boolean> {
+  // @ts-expect-error
+  const pathfinder: Pathfinder = bot.pathfinder
+  const goto = promisify(pathfinder.goto)
 
-  tryNextChest()
+  await goto(new goals.GoalBlock(location.x, location.y, location.z))
```

Comment:
[GoalBlock](https://github.com/PrismarineJS/mineflayer-pathfinder/blob/1b8b3d70ea40b888d36a74954f72b25ab9dedbcb/lib/goals.js#L24) aims to place the bot in the given location. Using [GoalGetToBlock](https://github.com/PrismarineJS/mineflayer-pathfinder/blob/1b8b3d70ea40b888d36a74954f72b25ab9dedbcb/lib/goals.js#L106) will allow the bot to approach the chest, end the goal when near, then interact with it.